lib/jsonapi/utils/underscore.ex

```diff
@@ -1,110 +1,30 @@
 defmodule JSONAPI.Utils.Underscore do
   @moduledoc """
-  Helpers for replacing underscores with dashes.
+  DEPRECATED. Please Use `JSONAPI.Utils.StringManipulation` instead.
   """
 
-  def underscore?, do: Application.get_env(:jsonapi, :underscore_to_dash, false)
+  @deprecated "Use JSONAPI.Utils.string.dasherize?/0 instead"
+  def underscore? do
+    JSONAPI.Utils.StringManipulation.dasherize?()
+  end
 
   @doc """
   Replace dashes between words in `value` with underscores
 
   Ignores dashes that are not between letters/numbers
-
-  ## Examples
-
-      iex> dash("top-posts")
-      "top_posts"
-
-      iex> dash("-top-posts")
-      "-top_posts"
-
-      iex> dash("-top--posts-")
-      "-top--posts-"
-
-      iex> dash(%{"foo-bar" => "baz"})
-      %{"foo_bar" => "baz"}
-
-      iex> dash({"foo-bar", "dollar-sol"})
-      {"foo_bar", "dollar-sol"}
-
-      iex> dash({"foo-bar", %{"a-d" => "z-8"}})
-      {"foo_bar", %{"a_d" => "z-8"}}
-
-      iex> dash(%{"f-b" => %{"a-d" => "z"}, "c-d" => "e"})
-      %{"f_b" => %{"a_d" => "z"}, "c_d" => "e"}
-
-      iex> dash(:"foo-bar")
-      :foo_bar
-
-      iex> dash(%{"f-b" => "a-d"})
-      %{"f_b" => "a-d"}
   """
-  def dash(value) when is_binary(value) do
-    String.replace(value, ~r/([a-zA-Z0-9])-([a-zA-Z0-9])/, "\\1_\\2")
-  end
-
-  def dash(map) when is_map(map) do
-    Enum.into(map, %{}, &dash/1)
-  end
-
-  def dash({key, value}) when is_map(value) do
-    {dash(key), dash(value)}
-  end
-
-  def dash({key, value}) do
-    {dash(key), value}
-  end
-
-  def dash(value) when is_atom(value) do
-    value
-    |> to_string()
-    |> dash()
-    |> String.to_atom()
-  end
-
+  @deprecated "Use JSONAPI.Utils.StringManipulation.underscore/1 instead"
   def dash(value) do
-    value
+    JSONAPI.Utils.StringManipulation.underscore(value)
```

Comment:
I'm not too concerned using `alias` because they can always use `:as`. 

I think we should go with `Utils.String` or we could with something like `Utils.Params` since this is a set of utilities for working on the param strings; `StringManipulation` is a handful.

Reply:
Done!